client.js

```diff
@@ -0,0 +1,291 @@
+                        } else { // otherwise no
+                            ok = false
+                        }
+                    } else { // if is not the previous 2 cases it means is not correct
+                        ok = false
+                    }
+
+                    var t1 = performance.now()
+                    resolve([ok, parseInt(t1 - start_request), num_callback])
+                })
+            }
+        )
+        .on("error", (error) => {
+            reject(error.message)
+        })
+    })
+        .then((data) => {
+            if (data[0] === false) {
+                if (performance.now() - start_request >= 300000) { // 5 minutes 
+                    // exceeded the double of the average to find the id of the video
+                    return data
+                }
+                return searchIdAndFeatures(title, id_video, description, ip, country, is_public, start, start_request, num_callback + 1)
+            } else {
+                // query name, country, result search, time from start to end of the request, time from start function call to end request
+                if (socket_localhost !== null) {
+                    socket_localhost.emit("update-graphs", country, data[0] === true, data[1], parseInt(performance.now() - start))
+                }
+                return data
+            }
+        })
+        .catch((err) => {
+            console.error(err)
+        })
+}
+
+async function searchId(title, id_video, ip, country, start, start_request, num_callback = 0) {
+    return new Promise((resolve, reject) => {
+        https.get(ip + "results?search_query=" + title,
+            { headers: { host: "www.youtube.com" } },
+            (res) => {
+                res.setEncoding("utf8")
+                var data = ""
+                res.on("data", function (chunk) {
+                    data += chunk
+                })
+                res.on("end", function () {
+                    let id_found = false
+                    if (data.includes(id_video)) { // check if the id is inside the result found
+                        id_found = true
+                    }
+                    var t1 = performance.now()
+                    resolve([id_found, parseInt(t1 - start_request), num_callback])
+                })
+            }
+        )
+        .on("error", (error) => {
+            reject(error.message)
+        })
+    })
+        .then((data) => {
+            if (data[0] === false) {
+                if (performance.now() - start_request >= 120000) { // 2 minutes 
+                    // exceeded the double of the average to find the id of the video
+                    return data
+                }
+                return searchId(title, id_video, ip, country, start, start_request, num_callback + 1)
+            } else {
+                // query name, country, result search, time from start to end of the request, time from start function call to end request
```

Comment:
Usually methods that accept multiple parameters can be made more transparent using a notation similar to this:

```c++
some_function(/*hostname=*/"localhost", /*port=*/8000, ...)
```

I know python allows setting the parameters explicitly (`some_function(hostname="localhost", ...)`) but I'm not sure about the javascript approach.